nucliadb_ingest/nucliadb_ingest/orm/resource.py

```diff
@@ -166,9 +166,28 @@ async def set_basic(self, payload: PBBasic, slug: Optional[str] = None):
         await self.get_basic()
         if self.basic is not None and self.basic != payload:
             self.basic.MergeFrom(payload)
+
             # We force the usermetadata classification to be the one defined
             if payload.HasField("usermetadata"):
                 self.basic.usermetadata.CopyFrom(payload.usermetadata)
+
+            if len(payload.fieldmetadata):
+                # keep only the last fieldmetadata item for each field. API
+                # users are responsible to manage fieldmetadata properly
+                fields = []
+                positions = {}
+                for i, fieldmetadata in enumerate(self.basic.fieldmetadata):
+                    field_id = self.generate_field_id(fieldmetadata.field)
+                    if field_id not in fields:
+                        fields.append(field_id)
+                    positions[field_id] = i
+
+                updated = [
+                    self.basic.fieldmetadata[positions[field]] for field in fields
+                ]
+
+                del self.basic.fieldmetadata[:]
+                self.basic.fieldmetadata.extend(updated)
```

Comment:
That's because assignment is now allowed to a repeated field like this. You need to use methods for repeated fields (like `add`, `append` or `extend`). So I need to clear it first and then extend with the updated list